### PR TITLE
Helpers - Tests and Structural Improvements

### PR DESCRIPTION
**Pull request by pixeldesu**
Kinda felt like doing this.

Split the Bootstrap and Title-related helpers into their own files, without altering any code about them. And while doing that, I also decided to write tests for the methods that weren't tested yet.

### CODE REVIEW

File: spec/helpers/application_helper/graph_methods_spec.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        allow(APP_CONFIG).to receive(:[]).with('site_name').and_return('pineapplespring')
+        expect(subject).to eq(<<-EOS.chomp)
+<meta property="og:title" content="Cunes" />
+<meta property="og:type" content="profile" />
+<meta property="og:image" content="http://test.host/images/large/no_avatar.png" />
+<meta property="og:url" content="http://test.host/raccoons" />
+<meta property="og:description" content="A bunch of raccoons in a trenchcoat." />
+<meta property="og:site_name" content="pineapplespring" />
+<meta property="profile:username" content="raccoons" />
+EOS
```
Comment (nilsding) [inline spec/helpers/application_helper/graph_methods_spec.rb]: with `<<~EOS` the indentation 
```suggestion
        expect(subject).to eq(<<~EOS.chomp)
          <meta property="og:title" content="Cunes" />
          <meta property="og:type" content="profile" />
          <meta property="og:image" content="http://test.host/images/large/no_avatar.png" />
          <meta property="og:url" content="http://test.host/raccoons" />
          <meta property="og:description" content="A bunch of raccoons in a trenchcoat." />
          <meta property="og:site_name" content="pineapplespring" />
          <meta property="profile:username" content="raccoons" />
        EOS
```

File: spec/helpers/application_helper/graph_methods_spec.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+      subject { user_twitter_card(user) }
+      it 'should generate a matching OpenGraph structure for a user' do
+        expect(subject).to eq(<<-EOS.chomp)
+<meta name="twitter:card" content="summary" />
+<meta name="twitter:site" content="@retrospring" />
+<meta name="twitter:title" content="Ask me anything!" />
+<meta name="twitter:description" content="Ask raccoons anything on Retrospring" />
+<meta name="twitter:image" content="http://test.host/images/large/no_avatar.png" />
+EOS
```
Comment (nilsding) [inline spec/helpers/application_helper/graph_methods_spec.rb]:
```suggestion
        expect(subject).to eq(<<~EOS.chomp)
          <meta name="twitter:card" content="summary" />
          <meta name="twitter:site" content="@retrospring" />
          <meta name="twitter:title" content="Ask me anything!" />
          <meta name="twitter:description" content="Ask raccoons anything on Retrospring" />
          <meta name="twitter:image" content="http://test.host/images/large/no_avatar.png" />
        EOS
```

File: spec/helpers/application_helper/graph_methods_spec.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      it 'should generate a matching OpenGraph structure for a user' do
+        allow(APP_CONFIG).to receive(:[]).with('site_name').and_return('pineapplespring')
+        expect(subject).to eq(<<-EOS.chomp)
+<meta property="og:title" content="raccoons answered: #{answer.question.content}" />
+<meta property="og:type" content="article" />
+<meta property="og:image" content="http://test.host/images/large/no_avatar.png" />
+<meta property="og:url" content="http://test.host/raccoons/a/#{answer.id}" />
+<meta property="og:description" content="#{answer.content}" />
+<meta property="og:site_name" content="pineapplespring" />
+EOS
```
Comment (nilsding) [inline spec/helpers/application_helper/graph_methods_spec.rb]:
```suggestion
        expect(subject).to eq(<<~EOS.chomp)
          <meta property="og:title" content="raccoons answered: #{answer.question.content}" />
          <meta property="og:type" content="article" />
          <meta property="og:image" content="http://test.host/images/large/no_avatar.png" />
          <meta property="og:url" content="http://test.host/raccoons/a/#{answer.id}" />
          <meta property="og:description" content="#{answer.content}" />
          <meta property="og:site_name" content="pineapplespring" />
        EOS
```

**Review by nilsding — APPROVED** [review]

File: app/helpers/application_helper/title_methods.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    end
+
+    list = [name]
+
+    list.push junction unless junction.nil?
+
+    unless content.nil?
+      content = strip_markdown(content)
+      if content.length > 45
+        content = content[0..42] + "..."
```
Comment (raccube) [inline app/helpers/application_helper/title_methods.rb]:
```suggestion
        content = content[0..42] + "…"
```

**Review by raccube — CHANGES_REQUESTED** [review]
👀

File: app/helpers/application_helper/title_methods.rb (hunk 4 above)
Comment (pixeldesu) [reply app/helpers/application_helper/title_methods.rb]: Of course I will accept a suggestion that will make the tests fail!

File: spec/helpers/application_helper/title_methods_spec.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    it 'should generate a proper title' do
+      expect(generate_title('Simon', 'says:', 'Nice!')).to eq('Simon says: Nice! | Waschmaschine')
+    end
+
+    it 'should only append a single quote to names that end with s' do
+      expect(generate_title('Andreas', 'says:', 'Cool!', true)).to eq('Andreas\' says: Cool! | Waschmaschine')
+    end
+
+    it 'should cut content that is too long' do
+      expect(generate_title('A title', 'with', 'a' * 50)).to eq('A title with aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa... | Waschmaschine')
```
Comment (raccube) [inline spec/helpers/application_helper/title_methods_spec.rb]:
```suggestion
      expect(generate_title('A title', 'with', 'a' * 50)).to eq('A title with aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa… | Waschmaschine')
```

**Review by raccube — CHANGES_REQUESTED** [review]

File: app/helpers/application_helper/title_methods.rb (hunk 4 above)
Comment (raccube) [reply app/helpers/application_helper/title_methods.rb]: Oops 🙈

**Review by raccube — APPROVED** [review]
🅱️asst

**Pull request merged**